sa/sa_test.go

```diff
@@ -991,6 +992,117 @@ func TestNewOrderAndAuthzs(t *testing.T) {
 	test.AssertDeepEquals(t, names, []string{"com.a", "com.b", "com.c", "com.d"})
 }
 
+func TestNewOrderAndAuthzs_MissingInnerOrder(t *testing.T) {
+	// An inner NewOrder object must exist or a nil pointer dereference gets
+	// thrown by the SA.
+
```

Comment:
We're not always good at doing this for test functions, but since you have a comment here anyway: make this a proper doc-comment. Put it above the function declaration, and have it start with the name of the function.

Also, the contents of this comment are no longer completely true: the SA won't panic on a nil pointer dereference, thanks to your change it will return errIncompleteRequest.